Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -1848,45 +1880,11 @@ namespace ScriptCanvasEditor
 
     void MainWindow::OnFileOpen()
     {
-        AZStd::string assetRoot;
-        {
-            AZStd::array<char, AZ::IO::MaxPathLength> assetRootChar;
-            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", assetRootChar.data(), assetRootChar.size());
-            assetRoot = assetRootChar.data();
-        }
-
-        AZStd::string assetPath = AZStd::string::format("%s/scriptcanvas", assetRoot.c_str());
-        QString filter;
-
-        AZStd::set<AZStd::string> filterSet { ".scriptcanvas" };
-
-        QStringList nameFilters;
-
-        QString globalFilter;
-        for (auto fileFilter : filterSet)
-        {
-            nameFilters.push_back(fileFilter.c_str());
-
-            AZStd::size_t filterStart = fileFilter.find_last_of("(");
-            AZStd::size_t filterEnd = fileFilter.find_last_of(")");
-
-            if (filterStart != AZStd::string::npos
-                && filterEnd != AZStd::string::npos)
-            {
-                AZStd::string substring = fileFilter.substr(filterStart + 1, (filterEnd - filterStart) - 1);
-
-                if (!globalFilter.isEmpty())
-                {
-                    globalFilter.append(" ");
-                }
-
-                globalFilter.append(substring.c_str());
-            }
-        }
+        AZ::IO::FixedMaxPath resolvedProjectRoot;
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath(resolvedProjectRoot, "@projectroot@");
 
-        globalFilter = QString("All ScriptCanvas Files (%1)").arg(globalFilter);
-
-        nameFilters.push_front(globalFilter);
+        const AZStd::string assetPath = AZStd::string::format("%s/scriptcanvas", resolvedProjectRoot.c_str());
```

Comment:
This should use AZ::IO::FixedMaxPath::operator/ for appending the paths together and `AZ::Utils::GetProjectPath` for retrieved the projectpath
```
const auto assetPath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "scriptcanvas";